syntax = "proto3";

package llm_engine;

// gRPC 상태나 빈 응답을 위해 import
import "google/protobuf/empty.proto";

// LLM Engine이 제공하는 서비스를 정의합니다.
service LLMService {
  // 클라이언트 스트리밍 RPC:
  // stt-service는 텍스트 청크 스트림을 보내고, llm-engine은 스트림 처리가
  // 완료된 후 단일 응답 메시지를 반환합니다.
  rpc ProcessTextStream(stream LLMStreamRequest) returns (LLMStreamSummary);
  // 만약 llm-engine이 별도의 응답을 줄 필요가 없다면 아래처럼 Empty를 반환해도 됩니다.
  // rpc ProcessTextStream(stream LLMStreamRequest) returns (google.protobuf.Empty);
}

// stt-service에서 llm-engine으로 보내는 메시지 (이전과 동일)
message LLMStreamRequest {
  string session_id = 1;
  string text_chunk = 2;
  bool is_final = 3;
}

// llm-engine이 클라이언트 스트림 종료 후 반환하는 단일 응답 메시지
message LLMStreamSummary {
  string session_id = 1;
  bool success = 2; // 전체 스트림 처리 성공 여부
  string message = 3; // 상태 메시지 또는 오류 상세 정보 (선택 사항)
  // 필요시 다른 요약 정보 필드 추가 가능
}